cdap-app-templates/cdap-etl/cdap-etl-batch/src/main/java/co/cask/cdap/etl/batch/spark/ETLSpark.java

```diff
@@ -114,11 +120,25 @@ public void beforeSubmit(SparkContext context) throws Exception {
       finishers.add(batchSink, sinkContext);
     }
 
+    Set<StageInfo> aggregators = phaseSpec.getPhase().getStagesOfType(BatchAggregator.PLUGIN_TYPE);
+    Integer numPartitions = null;
```

Comment:
I still need the null check later, since `aggregatorContext.getNumPartitions` returns null if nothing was set